### ISSUE TITLE
v11.1.0: TypeError: this.off is not a function, client.js line #4996

### ISSUE THREAD

**Issue by menturion** [matrix-org/matrix-js-sdk#1708]
matrix-js-sdk v11.1.0:

The following issue is thrown:
```

Getting saved sync failed TypeError: this.off is not a function
    at MatrixClient._startCallEventHandler (client.js:5067)
    at emitThree (events.js:131)
    at MatrixClient.emit (events.js:190)
    at SyncApi._updateSyncState (sync.js:1758)
    at SyncApi._syncFromCache (sync.js:757)
```
I think line #4996 in `client.js`  https://github.com/matrix-org/matrix-js-sdk/blob/acb9bc8cc5234326a7583514a8e120a4ac42eedc/src/client.js#L4996

should be replaced by 

`this.on("sync", this._startCallEventHandler);`

**Comment by t3chguy** [comment]
Nope, `off` is right - Here are its docs https://nodejs.org/api/events.html#events_emitter_off_eventname_listener 

Either your node version is too old or you're using an incompatible EventEmitter implementation

**Comment by menturion** [comment]
@t3chguy 

Thanks for your reply.

I updated Node `v12.13.1` to latest `v14.17.0`, cleared all caches and still getting this issue.

So, if nothing helps, I would have to replace `.off()` by `.removeListener()` by hand upon each Matrix JS update.

**Comment by t3chguy** [comment]
> Added in: v10.0.0

Make sure you don't have an nvm or some other symlinked node version being used instead as `.off` has been in since v10

**Comment by menturion** [comment]
I updated the LTS version to the latest version `16.2.0` due to the `.off()` alias has obvisouly been introduced in v16.2.0. 
Still getting `this.off is not a function`. 

Are there any advantages to use the `.off()` alias instead of using `.removeListener()`?

Wouldn't is make sense to specify a min required Node engine version by using ...

```
   "engines" : { 
      "node": ">= 16.2.0"
   },
   "engineStrict": true,
```

within the `package.json` in such a case?

**Comment by t3chguy** [comment]
As per my earlier message it was added in v10.

![image](https://user-images.githubusercontent.com/2403652/119479099-363a0880-bd48-11eb-9fb4-5fec6c778d5c.png)

I don't use v16 and it runs fine.

https://github.com/matrix-org/matrix-js-sdk#in-nodejs says Node 10+ for compat, and that aligns with the docs for `.off`

**Comment by menturion** [comment]
So according to this it should have been working fine for me too, because I was using Node` v12.13.1` and Matrix JS v11.0.0 without any problem. 

So something changed in Matrix JS v11.1.0 causing this issue.

**Comment by menturion** [comment]
@t3chguy 

I got it.

I referenced a version of browserify (`"browserify": "^16.5.2"`) which does not define the EventEmitter alias `.off()` in `browserify/events`.

I had to use the latest `v17.0.0` of browserify which defines the alias (here https://github.com/browserify/events/blob/48e3d18659caf72d94d319871106f089bb40002d/events.js#L321) ...

`EventEmitter.prototype.off = EventEmitter.prototype.removeListener;`

Now it works.
